daskms/table_proxy.py

```diff
@@ -194,44 +235,89 @@ def taql_factory(query, style='Python', tables=(), readonly=True):
             t.unlock()
 
 
-def _nolock_runner(table_future, fn, args, kwargs):
-    try:
-        return fn(table_future.result(), *args, **kwargs)
-    except Exception:
-        if logging.DEBUG >= log.getEffectiveLevel():
-            log.exception("Exception in %s", fn.__name__)
-        raise
-
-
-def _readlock_runner(table_future, fn, args, kwargs):
-    table = table_future.result()
-    table.lock(write=False)
-
-    try:
-        return fn(table, *args, **kwargs)
-    except Exception:
-        if logging.DEBUG >= log.getEffectiveLevel():
-            log.exception("Exception in %s", fn.__name__)
-        raise
-    finally:
-        table.unlock()
-
-
-def _writelock_runner(table_future, fn, args, kwargs):
-    table = table_future.result()
-    table.lock(write=True)
-
-    try:
-        result = fn(table, *args, **kwargs)
-        table.flush()
-    except Exception:
-        if logging.DEBUG >= log.getEffectiveLevel():
-            log.exception("Exception in %s", fn.__name__)
-        raise
-    else:
-        return result
-    finally:
-        table.unlock()
+def _nolock_runner(fn):
+    """
+    _nolock_runner wrapper with profiling
+    """
+    @wraps(fn)
```

Comment:
You've used @wraps as in the stackoverflow example, but this isn't fully necessary. Can you restore the original function and place the timing calls in that function.

Reply:
You should be able to do things like

```python
_nolock_runner.callls +=1
```

Reply:
So remove the function decorator for the wrapper, `wraps` 

Because of the need to keep track of how many times the function is being called,  I still need the wrapper function and have `_readlock_runner.calls` and `_readlock_runner.run_time` initialised on every new call. 

```
def _readlock_runner(table_future, fn, args, kwargs):
    """
    _readlock_runner wrapper with profiling
    """
    def wrapper(table_future, *args, **kwargs)
        table = table_future.result()
        table.lock(write=False)

        try:
            _readlock_runner.calls += 1
            start_time = time()
            result = fn(table_future.result(), *args, **kwargs)
            end_time = time()
            _readlock_runner.run_time.append(end_time - start_time)
            _function_runs[fn.__name__] = (_readlock_runner.run_time, _readlock_runner.calls)
            return result
        except Exception:
            if logging.DEBUG >= log.getEffectiveLevel():
                log.exception("Exception in %s", fn.__name__)
            raise
        finally:
            table.unlock()

    _readlock_runner.calls = 0
    _readlock_runner.run_time = []
    return wrapper
```

Reply:
How about this, @smasoka?

````python
def _readlock_runner(table_future, fn, args, kwargs):
    """
    _readlock_runner wrapper with profiling
    """
    table = table_future.result()
    table.lock(write=False)

    try:
        _readlock_runner.calls += 1
        start_time = time()
        result = fn(table_future.result(), *args, **kwargs)
        end_time = time()
        _readlock_runner.run_time.append(end_time - start_time)
        _function_runs[fn.__name__] = (_readlock_runner.run_time, _readlock_runner.calls)
        return result
    except Exception:
        if logging.DEBUG >= log.getEffectiveLevel():
            log.exception("Exception in %s", fn.__name__)
        raise
    finally:
        table.unlock()

_readlock_runner.calls = 0
_readlock_runner.run_time = []
```

Reply:
Now I don't even want to say what I was being confused about, its a none issue now.